app/helpers/email_signup_helper.rb

```diff
@@ -2,6 +2,8 @@ module EmailSignupHelper
   def email_signup_path(atom_feed_url)
     if EmailSignupPagesFinder.exists_for_atom_feed?(atom_feed_url)
       organisation_email_signup_information_path_from_atom_feed(atom_feed_url)
+    elsif URI.parse(atom_feed_url).path.match(%r{^/world/(.*)\.atom$})
```

Comment:
Do we need to rescue a `URI::InvalidURIError` like we do in WorldLocationEmailSignup model? not sure if this is user input or not.

Reply:
Hmmm I don't think so as the atom_feed_url in most cases (where I've looked) comes from a method call on an instance from the db and most of the time it would bomb before getting to this point however because this method is used indirectly in quite a few places I've added the rescue which points the user to the homepage if it's invalid as a precaution.